demo/src/main/java/com/example/demo/services/impl/BalanceServiceImpl.java

```diff
@@ -1,35 +1,34 @@
 package com.example.demo.services.impl;
 
-import com.example.demo.controller.BalanceController;
+import com.example.demo.dto.balance.BalanceInfoDto;
 import com.example.demo.entities.Balance;
 import com.example.demo.repository.BalanceRepository;
-import com.example.demo.repository.TransactionRepository;
 import com.example.demo.services.BalanceService;
 import lombok.RequiredArgsConstructor;
 import org.springframework.stereotype.Service;
 
-import java.util.Date;
+import java.util.List;
+import java.util.stream.Collectors;
 
 @Service
-
+@RequiredArgsConstructor
 public class BalanceServiceImpl implements BalanceService {
+
     private final BalanceRepository balanceRepository;
-    private final TransactionRepository transactionRepository;
 
-    public BalanceServiceImpl(BalanceRepository balanceRepository, TransactionRepository transactionRepository) {
-        this.balanceRepository = balanceRepository;
-        this.transactionRepository = transactionRepository;
+    private BalanceInfoDto mapToBalanceDto(Balance balance){
+        BalanceInfoDto dto = new BalanceInfoDto();
+        dto.setClientId(balance.getId());
+        dto.setLastTransactionDate(balance.getLastTransactionDate());
+        dto.setBalanceAmount(balance.getBalanceAmount());
+        return dto;
     }
 
     @Override
-    public void saveCurrentBalance(Balance balance) {
-
-
-        Double sumOfClientTransactionAmounts = transactionRepository.calculateSumOfClientTransactionAmounts(balance.getClient().getId());
```

Comment:
Double sum = transactionRepository.calculateSumOfClientTransactionAmounts(balance.getClient().getId()); => method name is too long, plz try to maintain method names as short as possible. Method name is builded by two main parameters: 1. what method do, 2. what it needs for proper execution. For example: calculateSumOfClientTransactionAmounts => transactionsSumByClientId